docs/developer/extending/apps/building-payment-app.mdx

```diff
@@ -0,0 +1,856 @@
+---
+title: How to Build a Payment App
+---
+
+## Overview
+
+This tutorial will guide you through creating a basic Payment App with [Transactions API](developer/extending/webhooks/synchronous-events/transaction.mdx).
+
+:::note
+
+We will integrate with a fictional **Dummy Payment Gateway**. This gateway will simulate real payment processing without handling actual payments.
+
+:::
```

Comment:
I would add here a mermaid chart that would show the full flow that we will achieve with this section. 
The chart would include storefront, saleor, apps, mutations that we use, webhooks that we use. This should give you an overall view of the final flow that we want to implement.